### PR TITLE
DAOS-14475 cq: fix find_base issue from #13135

### PR DESCRIPTION
**Pull request by daltonbohning**
- Need to reference $ORIGIN/$base instead of $base
- Need to redirect stderr and stdout because the parent script uses set -eu

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: utils/githooks/find_base.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ else
     TARGET="$ORIGIN/master"
     min_diff=-1
     for base in "${all_bases[@]}"; do
-        git rev-parse --verify "$base" 2&>/dev/null || continue
+        _=$(git rev-parse --verify "$ORIGIN/$base") || continue
```
Comment (brianjmurrell) [inline utils/githooks/find_base.sh:41]: What exactly is the issue here that you are junking the output into `$_`?

Comment (daltonbohning) [reply utils/githooks/find_base.sh:41]: Definitely open to bash suggestions. We don't get along :)

Comment (brianjmurrell) [reply utils/githooks/find_base.sh:41]: So the relationship between this and a caller is that if a caller is using `set -e` and does not check the return of this script'S execution in any way, any `!0` exit of this script will cause the caller to take it's `ERROR` handling path (i.e. exit immediately or call the `ERROR` `trap` if one is set).

I suspect what you are looking for is this:

```suggestion
        git rev-parse --verify "$ORIGIN/$base" &> /dev/null || continue
```

to discard both `stdout` and `stderr`.

Comment (daltonbohning) [reply utils/githooks/find_base.sh:41]: Ah, yes! Using `2&>` - not correct
```
$ git rev-parse --verify origin/foo 2&> /dev/null || echo hello
hello
$ git rev-parse --verify origin/master 2&> /dev/null || echo hello
hello
```
using `&>` - correct
```
$ git rev-parse --verify origin/foo &> /dev/null || echo hello
hello
$ git rev-parse --verify origin/master &> /dev/null || echo hello
# no output
```
Thanks!

**Review by brianjmurrell — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by daltonbohning** [comment]
Please use the PR description for merge
```
Need to reference $ORIGIN/$base instead of $base
Need to redirect stderr and stdout because the parent script uses set -eu
```

**Pull request merged**